pkg/controllers/allocation/filter.go

```diff
@@ -88,14 +88,11 @@ func (f *Filter) hasSupportedSchedulingConstraints(pod *v1.Pod) error {
 }
 
 func (f *Filter) matchesProvisioner(pod *v1.Pod, provisioner *v1alpha3.Provisioner) error {
-	if pod.Spec.NodeSelector == nil {
-		return nil
-	}
 	name, ok := pod.Spec.NodeSelector[v1alpha3.ProvisionerNameLabelKey]
-	if !ok {
+	if ok && name == provisioner.Name {
 		return nil
 	}
-	if name == provisioner.Name {
+	if !ok && v1alpha3.DefaultProvisioner.Name == provisioner.Name {
```

Comment:
If we invert the predicate it read a bit more smoothly

```
if ok && provisioner.Name == name
if !ok && provisioner.Name == v1alpha3.DefaultProvisioner.Name
```
If we could somehow assume that provisioner.Name was always populated, we could even remove the `ok` bit.

Reply:
I'm not sure it's worth assuming that. Maybe we could, but feels like something that could be forgotten at some point. IMO the ok isn't too dirty and we should probably just leave it.